tools/mock-service-host/src/mid/responser.ts

```diff
@@ -228,18 +229,48 @@ export class ResponseGenerator {
         }
     }
 
-    public async generate(
-        operation: Operation,
-        config: Config,
+    private validateExampleResponse(
+        liveValidator: oav.LiveValidator,
         liveRequest: LiveRequest,
-        lroCallback: string | null
+        example: SwaggerExample
     ) {
+        for (const statusCode in example.responses) {
+            // exception will raise if is not valid example responses
+            liveValidator.validateLiveResponse(
```

Comment:
Do we need to verify all responses are correct?  What if we want to return response of 200 and that one is correct, but another response in example is wrong?

Reply:
Can we check the correctness based on the specific status code?

Reply:
The status code is chosen after the responses is generated: https://github.com/Azure/azure-sdk-tools/blob/main/tools/mock-service-host/src/mid/coordinator.ts#L296

let me try to do some refactor to move this logic ahead to this point.

Reply:
Hi @lirenhe , @tadelesh , I refactored to validate example and mock response for specific status code: https://github.com/Azure/azure-sdk-tools/pull/3372/files#diff-35cd40dd5e37c804d5dc541a0528c0fce7e0f71c90878a5494e69511840c5d86R335